types/chrome-har-capturer/index.d.ts

```diff
@@ -0,0 +1,168 @@
+/// <reference types="node" />
+
+import { EventEmitter } from "events";
+
+export interface Page<U> {
+    id: string;
+    title: any;
+    startedDateTime: string;
+    pageTimings: {
+        onContentLoad: number;
+        onLoad: number;
+    };
+    _user: U;
+}
+
+export interface Entry {
+    pageref: string;
+    startedDateTime: string;
+    time: number;
+    request: {
+        method: any;
+        url: any;
+        httpVersion: any;
+        cookies: never[];
+        headers: Array<{
+            name: string;
+            value: any;
+        }>;
+        queryString: Array<{
+            name: string;
+            value: any;
+        }>;
+        headersSize: number;
+        bodySize: number;
+        postData: {
+            mimeType: any;
+            params: Array<{
+                name: string;
+                value: any;
+            }>;
+            text: any;
+        } | undefined;
+    };
+    response: {
+        status: any;
+        statusText: any;
+        httpVersion: any;
+        cookies: never[];
+        headers: Array<{
+            name: string;
+            value: any;
+        }>;
+        redirectURL: any;
+        headersSize: number;
+        bodySize: number;
+        _transferSize: any;
+        content: {
+            size: any;
+            mimeType: any;
+            compression: number | undefined;
+            text: any;
+            encoding: string | undefined;
+        };
+    };
+    cache: {};
+    _fromDiskCache: any;
+    timings: {
+        blocked: any;
+        dns: number;
+        connect: number;
+        send: number;
+        wait: number;
+        receive: number;
+        ssl: number;
+    };
+    serverIPAddress: any;
+    connection: string;
+    _initiator: any;
+    _priority: any;
+    _webSocketMessages: any;
+    _resourceType: any;
+}
+
+export interface HAR<U> {
+    log: {
+        version: string;
+        creator: {
+            name: string;
+            version: string;
+            comment: string;
+        };
+        pages: Array<Page<U>>;
+        entries: Array<(Entry | null)>;
+    };
+}
+
+export type LoaderEvents = "load" | "done" | "fail" | "har";
+export interface LoaderEventCallback<U> {
+    "load": (url: string, index: number, urls: readonly string[]) => void;
+    "done": (url: string, index: number, urls: readonly string[]) => void;
+    "fail": (url: string, err: any, index: number, urls: readonly string[]) => void;
+    "har": (har: HAR<U>) => void;
+}
+
+export interface RunOptions<T> {
+    /** Chrome Debugging Protocol host. Defaults to localhost */
+    host?: string;
```

Comment:
I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)